### DISCUSSION TITLE
courses/computer-structure/%EC%9E%90%EB%A3%8C%EA%B5%AC%EC%A1%B0/%ED%81%90%EC%99%80%EC%8A%A4%ED%83%9D

### DISCUSSION THREAD

**Discussion by giscus** [crwdu/lms-contents-js101#72] [Q&A]
# courses/computer-structure/%EC%9E%90%EB%A3%8C%EA%B5%AC%EC%A1%B0/%ED%81%90%EC%99%80%EC%8A%A4%ED%83%9D

https://lms.vanillacoding.co/courses/computer-structure/%EC%9E%90%EB%A3%8C%EA%B5%AC%EC%A1%B0/%ED%81%90%EC%99%80%EC%8A%A4%ED%83%9D

**Comment by smileboy0014** [comment]
### 2개의 큐로 스택을 구현해보세요

- 초기화
1. Queue1(q1) 과 Queue2(q2)를 정의한다.

- push
1. q1에 요소를 enqueue 한다.

- pop
1. 만약에 q1이 비었다면 비었다는 것을 리턴한다.
2. q1에 있는 요소가 하나 남을때까지 다음을 반복한다.
    1. q1에서 요소를 dequeue 한다.
    2. q2에 요소를 enqueue 한다.
3. q1에서 dequeue를 하고, dequeue한 요소를 리턴한다.
4. q1과 q2를 바꾼다.
    
      
    

### 2개의 스택으로 큐를 구현해보세요.

- 초기화
1. stack1(stk1)과 stack2(stk2)를 정의한다.

- enqueue
1. stk1에 요소를 push 한다.

- dequeue
1. 만약 stk2의 사이즈가 0이라면
    1. 만약 stk1의 사이즈가 0이라면 비었다는 것을 리턴한다.
2. 만약 stk2의 사이즈가 0이 아니라면
    1. stk2에서 pop을 하고, pop한 요소를 리턴한다.
3. stk1의 사이즈가 0이 될때까지 다음을 반복한다.
    1. stk1에서 pop 한다.
    2. stk2에 push 한다.
4. stk2에서 pop을 하고, pop한 요소를 리턴한다.

**Reply by kam1989** [reply]
3: 괄호 문제

괄호 왼쪽 목록 배열 작성
괄호 오른쪽 목록 배열 작성

문자열 목록 순회{
	if 괄호 왼쪽{
		스택에 입력
	}
	else if 괄호 오른쪽
	{
		스택 맨 위 peek 
		if 괄호 왼쪽 오른쪽 쌍 일치{
				스택 pop	
		}
		else if 괄호 왼쪽 오른쪽 쌍 불일치{
				return 불일치 확인	
		}
	}
}

**Comment by kam1989** [comment]
3: 괄호 일치 확인

괄호 왼쪽 목록 배열 작성
괄호 오른쪽 목록 배열 작성

문자열 목록 순회{
-if 괄호 왼쪽{
--스택에 입력
-}
-else if 괄호 오른쪽{
--스택 맨 위 peek 
--if 괄호 왼쪽 오른쪽 쌍 일치{
---스택 pop	
--}
--else if 괄호 왼쪽 오른쪽 쌍 불일치{
---return 불일치 확인	
--}
-}
}
